cirq/ops/controlled_gate.py

```diff
@@ -68,23 +125,27 @@ def _unitary_(self) -> Union[np.ndarray, NotImplementedType]:
         sub_matrix = protocols.unitary(self.sub_gate, None)
         if sub_matrix is None:
             return NotImplemented
-        return linalg.block_diag(np.eye(sub_matrix.shape[0]), sub_matrix)
+        return linalg.block_diag(
+                    np.eye(sub_matrix.shape[0] << (self.num_controls - 1)),
+                    sub_matrix)
```

Comment:
This is where it goes wrong. PTAL at the failing tests to see what cases break it.